modules/service/src/main/java/com/devonfw/module/service/common/base/client/AbstractServiceClientErrorFactory.java

```diff
@@ -71,10 +71,24 @@ public RuntimeException unmarshall(String data, String format, int statusCode, S
   private RuntimeException createException(Map<String, Object> map, String service) {
 
     String code = (String) map.get(ServiceConstants.KEY_CODE);
+    if (isEmpty(code)) {
+      code = map.get(ServiceConstants.KEY_STATUS).toString();
```

Comment:
this could yield to a NPE!?

Reply:
??? Your comment does not make sense to me. That is exactly the purpose of the `isEmtpy` method to prevent NPE. Did you read the code?

Reply:
`map.get(ServiceConstants.KEY_STATUS)` could be null right?

Reply:
Ouch. Sure. Sorry, I was stupid and missed this. Will fix this tomorrow. Thanks...